.github/workflows/windows-artifacts.yml

```diff
@@ -23,85 +23,83 @@ jobs:
+      uses: actions/checkout@v4
 
-  build-e2e:
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: ${{ matrix.go }}
+
+    - name: Set path for heat.exe and light.exe
+      shell: bash
+      run: echo "$WIX\\bin" >>$GITHUB_PATH
       
-    steps:
-      - name: Check out repository code
-        uses: actions/checkout@v4
+    - name: Build Windows installer
+      run: make out/windows-amd64/crc-windows-installer.zip
       
-      - name: Build and archive e2e image
-        # use github.sha as ID to correlate various workflows triggered by the same event
-        run: |
-          CRC_E2E_IMG_VERSION=gh make containerized_e2e
-          podman save -o ${{ env.IMAGE_NAME_E2E }}.tar quay.io/crcont/${{ env.IMAGE_NAME_E2E}}:gh
-
-      - name: Upload e2e image
-        uses: actions/upload-artifact@v4
-        with:
-          name: ${{ env.IMAGE_NAME_E2E }}-gh
-          path: ${{ env.IMAGE_NAME_E2E }}.tar
+    - name: Upload windows installer artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: windows-installer
+        path: "./out/windows-amd64/crc-windows-installer.zip"
 
-  build-integration:
+  build-qe:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
-    
-    steps:
-      - name: Check out repository code
-        uses: actions/checkout@v4
-      
-      - name: Build and archive integration image
-        # use github.sha as ID to correlate various workflows triggered by the same event
-        run: |
-          CRC_INTEGRATION_IMG_VERSION=gh make containerized_integration
-          podman save -o ${{ env.IMAGE_NAME_INTEGRATION }}.tar quay.io/crcont/${{ env.IMAGE_NAME_INTEGRATION }}:gh
+      matrix:
+        os: ['darwin','windows', 'linux']
+        arch: ['amd64', 'arm64']
+        exclude:
+        - arch: 'arm64'
+          os: 'windows'
+        - arch: 'arm64'
+          os: 'linux'
       
-      - name: Upload integration image
-        uses: actions/upload-artifact@v4
-        with:
-          name: ${{ env.IMAGE_NAME_INTEGRATION }}-gh
-          path: ${{ env.IMAGE_NAME_INTEGRATION }}.tar
+    steps:
+    - name: Check out repository code
+      uses: actions/checkout@v4
+    
+    - name: Build qe oci images ${{matrix.os}}-${{matrix.arch}}
+      run: |
+        # e2e
+        CRC_E2E_IMG_VERSION=gh ARCH=${{matrix.arch}} OS=${{matrix.os}} make containerized_e2e
+        podman save -o ${{ env.IMAGE_NAME_E2E }}-${{matrix.os}}-${{matrix.arch}}.tar \
+          quay.io/crcont/${{ env.IMAGE_NAME_E2E}}:gh-${{matrix.os}}-${{matrix.arch}}
+        # integration
+        CRC_INTEGRATION_IMG_VERSION=gh ARCH=${{matrix.arch}} OS=${{matrix.os}} make containerized_integration
+        podman save -o ${{ env.IMAGE_NAME_INTEGRATION }}-${{matrix.os}}-${{matrix.arch}}.tar \
+          quay.io/crcont/${{ env.IMAGE_NAME_INTEGRATION}}:gh-${{matrix.os}}-${{matrix.arch}}
```

Comment:
So we are building the container image for e2e and integration and then we are saving the image for same using quay.io using `podman save` why?

Reply:
We are building the images but not pushing them, instead we are saving them as flat files which are uploaded as artifacts to the action.

Then on the workflowrun we are downloading those artifacts and load them as images back

Reply:
Ack